src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilder.cs

```diff
@@ -174,10 +174,14 @@ public class LockFileBuilder
                 // Check if warnings should be displayed for the current framework.
                 var tfi = project.GetTargetFramework(targetGraph.Framework);
 
-                var warnForImportsOnGraph = tfi.Warn
+                bool warnForImportsOnGraph = tfi.Warn
                     && (target.TargetFramework is FallbackFramework
                         || target.TargetFramework is AssetTargetFallbackFramework);
 
+                // Check if warnings should be shown for this targetFramework if a package is used with a FrameworkRef pulling in WPF or WinForms
```

Comment:
I was expecting us to implement these warnings in the SDK.  I don't think NuGet needs to have this logic.

Reply:
given that, my guess is that we'd prefer to have them out of nuget as well. @nkolev92 - thoughts?

Reply:
I think there's been some experience questions about Framework Reference and authoring packages specifically for certain ones and having NuGet validate them. https://github.com/NuGet/Home/issues/9119

While it'd be nice if NuGet can make as many compatibility decisions as possible at restore time, it's ok if we defer that for a future change.

Reply:
yes, let's keep framework ref out of NuGetFramework.
	Yep.
NU1703 will likely be asked for...
	nikolche says consider later in net5.0 or later...
	for now, ask sdk to raise that warning.
	if we added it back, we'd have them remove or dedup.
let's discuss more offline. but remove from 1st change.